### ISSUE TITLE
Data Destruction for ROI PA

### ISSUE THREAD

**Issue by mnataraj92** [episphere/connect#1179]
The Number of Available Physical Activity Reports variable (d_686238347_d_794047378) should be retained in the stub record upon data destruction. This has also been indicated in the data destruction SOP.

**Comment by brotzmanmj** [comment]
@mnataraj92 is this the only variable related to ROI Physical Activity that will be retained in the data destruction process? 

And can you also explain what this variable signifies? There isn't a description of it in the dictionary.

Is there another issue related to data destruction for ROI Physical Activity or is this the only one?

**Comment by mnataraj92** [comment]
Yes, Rohan confirmed with Brian that this will be the only ROI Physical Activity variable that will be retained upon data destruction. This variable is a numeric value (up to 99) showing the the number of available physical activity reports. This will be stored in the participants table. 

To my knowledge this is the only issue related to data destruction for ROI physical activity.

**Comment by rohanjay10** [comment]
Some additional info on the # of available reports variable as discussed with Brian: This variable show the count of previously available reports, that would be compared with the current count of available reports to determine if a new report is available. This would be initiated when a participants logs into the PWA and the new report available banner is shown.

**Comment by brotzmanmj** [comment]
@mnataraj92 please update the data dictionary description of this variable to include the more complete description of this variable. @rohanjay10  please confirm this variable pertains to all reports and not just ROI Physical Activity reports. In other words, when we return Genetic Ancestry report, it will also accrue to this variable?

For data destruction for ROI PA: We need a discussion and decision of what variables to keep. We decided we will destroy the ROI PA data in the ROI BQ table, since those data are derived from Module 2 which will also be destroyed. 

We need to retain: 
- One variable to indicate that the report was delivered to the ppt. A variable indicating a report was viewed by the ppt would also suffice. The options for this appear to be either: Report delivered flag in ROI BQ table(yes/no) (d_730304319) or Date ROI PA Report First Viewed (d_749055145) which in the Participants table. @bransteitterbr is there a preference as to which is retained from a technical functioning standpoint?
- Any variables that are needed such that things don't break. @bransteitterbr  can you advise on this as well?

Thanks

**Comment by rohanjay10** [comment]
HI @brotzmanmj  I confirmed with Brian that this variable would include all reports, including Genetic Ancestry report when available, and not just the Physical Activity Report.

**Comment by bransteitterbr** [comment]
@brotzmanmj  For my needs the two fields that I need retained are Number of Available Physical Activity Reports (d_794047378) and Flag for Report Unread/Viewed/Declined (d_446235715).  I don't need ROI BQ table(yes/no) (d_730304319) or Date ROI PA Report First Viewed (d_749055145)

**Comment by brotzmanmj** [comment]
@bransteitterbr  In that case, I propose that we should retain these three variables in the event of data destruction request:
Number of Available Physical Activity Reports (d_794047378)
Flag for Report Unread/Viewed/Declined (d_446235715)
Date ROI PA Report First Viewed (d_749055145)

And destroy all of the other variables created for the ROI PA in the Participants table and the ROI BQ table. 
@mnataraj92  please list those variables here so we have clarity on this.

@m-j-horner please confirm if you agree

**Comment by mnataraj92** [comment]
Variables to retain:
Number of Available Physical Activity Reports (d_794047378)
Flag for Report Unread/Viewed/Declined (d_446235715)
Date ROI PA Report First Viewed (d_749055145)

Variables to destroy: 
Autogenerated Date/Time for Report Declined (d_295732360)
Flag for Report Delivered (d_730304319)
Autogenerated Date/Time for Report Delivered (d_416831581)
True missing flag for those who skipped the raw base recreational activity questions (d_342646001)
Measure for engagement in muscle-strengthening activities (d_205380968)
Physical activity guideline comparison (d_449038410)

**Comment by jacobmpeters** [comment]
The `ROI.physical_activity` table is appended each time new participants become eligible for an ROI PA report. So even if the data are deleted upstream in Firestore, there is currently no mechanism to delete those data from the ROI.physical_activity table. If we want to delete any of the derived variables in the ROI.physical_activity table, I would need to develop a separate process for that. 

@mnataraj92 I haven't been involved in data destruction yet. Which variables would I use to trigger a data destruction process for the `ROI.physical_activity` table?

**Comment by jhflorey** [comment]
> Variables to retain: Number of Available Physical Activity Reports (d_794047378)- Participants Table Flag for Report Unread/Viewed/Declined (d_446235715)- Participants Table Date ROI PA Report First Viewed (d_749055145)- Participants Table
> 
> Variables to destroy: Autogenerated Date/Time for Report Declined (d_295732360)- Participants Table Autogenerated Date/Time for Report Delivered (d_416831581)- ROI BQ Table True missing flag for those who skipped the raw base recreational activity questions (d_342646001)- ROI BQ Table Measure for engagement in muscle-strengthening activities (d_205380968)- ROI BQ Table Physical activity guideline comparison (d_449038410)- ROI BQ Table

@rohanjay10 will we add the retained variables in this release or the next release?

**Comment by rohanjay10** [comment]
@jhflorey As Data Destruction for ROI PA is planned for next release I believe these retained variables will be added for the next release as well

**Comment by jhflorey** [comment]
@rohanjay10 got it. Thanks for your sharing.

**Comment by rohanjay10** [comment]
Hi @mnataraj92 , with the latest variable changes could you please provide the latest list of variables to retain and destroy

**Comment by mnataraj92** [comment]
Variables to retain:
Number of Available Physical Activity Reports [renaming to Numeric variable showing the # of available reports] (d_794047378, Firestore & Participants table)

Flag for Report Unread/Viewed/Declined (d_686238347d_446235715, Firestore & Participants table)

Date ROI PA Report First Viewed (d_686238347_d_749055145, Firestore & Participants table)

Variables to destroy:
Autogenerated Date/Time for Report Declined (d_686238347_d_295732360, Firestore & Participants table)

Autogenerated Date/Time for Report Delivered [renaming to Autogenerated date/time for ROI physical activity table appended](d_416831581, ROI BQ table)

Measure for engagement in muscle-strengthening activities (d_205380968, ROI BQ table)

Physical activity guideline comparison (d_449038410, ROI BQ table)

**Comment by jhflorey** [comment]
@jacobmpeters @rohanjay10 my code changes are ready for testing in dev.

**Comment by jacobmpeters** [comment]
@jhflorey @rohanjay10 My understanding is that Jessica will also be triggering my code to delete the `ROI.physical_activity` data in BigQuery. I am working on my code this afternoon and Jessica will need to trigger my code within hers. I'll let you know when I am done.

It would not make sense to test in dev until everything is complete and integrated.

**Comment by jhflorey** [comment]
@jacobmpeters my code change is to add new stub variables. i've not implement how to trigger your code in BigQuery

**Comment by jacobmpeters** [comment]
@jhflorey Thanks for explaining. I misunderstood. I am nearly done with my code. I'll ask you to review it when its done.

**Comment by jacobmpeters** [comment]
@rohanjay10 My PR for destruction of the ROI.physical_activity data in BQ is complete and pending review.

**Comment by jacobmpeters** [comment]
@rohanjay10 @mnataraj92 My data destruction code is in dev and ready for testing

**Comment by sonyekere** [comment]
Hello Everyone. @anthonypetersen  has confirmed that he will be able to assist by running the code needed for data destruction. @mnataraj92 please let him know when he should trigger this so testing can commence. Thank you!

**Comment by mnataraj92** [comment]
Tested data destruction in dev and this looks good!

**Comment by mnataraj92** [comment]
Tested data destruction in stage and this look as expected! Data destruction for ROI PA is ready for prod.
